engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/ProcessInstanceRestServiceInteractionTest.java

```diff
@@ -3634,6 +3640,254 @@ public void testAsyncProcessInstanceModificationCancellationSource() {
     verify(mockModificationBuilder).cancellationSourceExternal(true);
   }
 
+  @Test
+  public void shouldSetVariablesAsync() {
+    // given
+    Batch batchMock = createMockBatch();
+    when(runtimeServiceMock.setVariablesAsync(any(), any(), any(), any()))
+        .thenReturn(batchMock);
+
+    SetVariablesAsyncDto body = new SetVariablesAsyncDto();
+
+    VariableValueDto variableValueDto = new VariableValueDto();
+    variableValueDto.setValue("bar");
+
+    body.setVariables(Collections.singletonMap("foo", variableValueDto));
+
+    // when
+    Response response = given()
+          .contentType(ContentType.JSON)
+          .body(body)
+        .then().expect()
+          .statusCode(Status.OK.getStatusCode())
+        .when()
+          .post(PROCESS_INSTANCE_SET_VARIABLES_ASYNC_URL);
+
+    // then
+    ArgumentCaptor<Map> captor = ArgumentCaptor.forClass(Map.class);
+    verify(runtimeServiceMock).setVariablesAsync(
+        eq(null),
+        eq(null),
+        eq(null),
+        captor.capture()
+    );
+
+    Assertions.assertThat(captor.getValue().get("foo")).isEqualTo("bar");
+
+    verifyBatchJson(response.asString());
+  }
+
+  @Test
+  public void shouldThrowExceptionWhenSetVariablesAsync_UnsupportedType() {
+    // given
+    Batch batchMock = createMockBatch();
+    when(runtimeServiceMock.setVariablesAsync(any(), any(), any(), any()))
+        .thenReturn(batchMock);
+
+    SetVariablesAsyncDto body = new SetVariablesAsyncDto();
+
+    VariableValueDto variableValueDto = new VariableValueDto();
+    variableValueDto.setValue("bar");
+    variableValueDto.setType("unknown");
+
+    body.setVariables(Collections.singletonMap("foo", variableValueDto));
+
+    // when + then
+    given()
+          .contentType(ContentType.JSON)
+          .body(body)
+        .then().expect()
+          .statusCode(Status.BAD_REQUEST.getStatusCode())
+          .body(containsString("{\"type\":\"InvalidRequestException\"," +
+              "\"message\":\"Cannot set variables: Unsupported value type 'unknown'\"}"))
+        .when()
+          .post(PROCESS_INSTANCE_SET_VARIABLES_ASYNC_URL);
+  }
+
+  @Test
+  public void shouldThrowExceptionWhenSetVariablesAsync_JavaSerializationFormatProhibited() {
```

Comment:
I guess it is just fine like that, but what it really tests is that `ProcessEngineException`s are transformed to `InvalidRequestException`s, right?